docs/topics/multiplatform/multiplatform-discover-project.md

```diff
@@ -1,169 +1,315 @@
-plugins {
-    kotlin("multiplatform") version "%kotlinVersion%"
+fun greeting() {
+    println("Hello, Kotlin Multiplatform!")
 }
 ```
 
-</tab>
-<tab title="Groovy" group-key="groovy">
+Kotlin code shared among platforms is typically located in the `commonMain` directory. The location of code files is
+important, as it defines the list of platforms to which this code is compiled.
 
-```groovy
-plugins {
-    id 'org.jetbrains.kotlin.multiplatform' version '%kotlinVersion%'
-}
-```
+The Kotlin compiler gets the source code as input and produces a set of platform-specific binaries as a result. When
+compiling multiplatform projects, it can produce multiple binaries from the same code. For example, the compiler can
+produce JVM `.classfiles` and native executable files from the same Kotlin file:
+
+![Common code](common-code-diagram.svg){width=700}
+
+Not every piece of Kotlin code can be compiled to all platforms. The Kotlin compiler prevents you from using
+platform-specific functions or classes in your common code since this code can't be compiled to a different platform.
 
-</tab>
-</tabs>
+For instance, you can't use the `java.io.File` dependency from the common code because it can't be compiled to native
+platforms:
 
-The `kotlin-multiplatform` plugin configures the project for creating an application or library to work on multiple platforms 
-and prepares it for building on these platforms. 
+![Unresolved Java reference](unresolved-java-reference.png){width=500}
 
-In the file `build.gradle(.kts`), it creates the `kotlin` extension at the top level, which includes 
-configuration for [targets](#targets), [source sets](#source-sets), and dependencies.
+In common code, you can use Kotlin Multiplatform libraries. These libraries provide a common API that can be implemented
+differently on different platforms. In this case, platform-specific APIs serve as extra parts, and trying to use such an
+API in common code results in an error.
+
+For example, `kotlinx.coroutines` is a Kotlin Multiplatform library that supports all targets, but it also has a
+platform-specific part that converts `kotlinx.coroutines` concurrent primitives to JDK concurrent primitives,
+like `fun CoroutinesDispatcher.asExecutor(): Executor`. This additional part of the API isn't available in `commonMain`.
 
 ## Targets
 
-A multiplatform project is aimed at multiple platforms that are represented by different targets. A target is part of the 
-build that is responsible for building, testing, and packaging the application for a specific platform, such as macOS, 
-iOS, or Android. See the list of [supported platforms](multiplatform-dsl-reference.md#targets).
+Targets define the platforms to which Kotlin compiles the common code. These could be, for example, the JVM, JS,
+Android, iOS, or Linux. The previous example compiled the common code to the JVM and native targets. This is how the
+Kotlin compiler understands that you want to compile to those specific platforms.
```

Comment:
> This is how the
Kotlin compiler understands that you want to compile to those specific platforms

"How Kotlin compiler understands" is not explained at this point yet. Answer to "How" is given below, by explaining that a Kotlin Target is an identifier, that you have to "declare" targets and that to "declare" targets you're writing something like `jvm()` in build script. 

I'd suggest to remove this sentence